client/src/components/Results/index.js

```diff
@@ -1,108 +1,101 @@
-          reject(error);
-        });
-    });
-    return promise;
-  }
-
-  componentDidMount() {
-    this.getData().then((json) => {
+export default (Results) => {
+  const [chartData, setChartData] = useState(null);
+  useEffect(() => {
+    getData().then((json) => {
       json = json['Time Series (Daily)'];
       const labels = Object.keys(json);
       const values = Object.values(json);
       const data = values.map((x) => parseFloat(x['5. adjusted close']));
       console.log(labels);
-      const chartData = {
+      const chart = {
         labels: labels.reverse(),
-        datasets: [
-          {
-            label: 'AAPL stock price',
-            fill: false,
-            redraw: true,
-            lineTension: 0.1,
-            backgroundColor: 'rgba(75,192,192,0.4)',
-            borderColor: 'rgba(75,192,192,1)',
-            borderCapStyle: 'butt',
-            borderDash: [],
-            borderDashOffset: 0.0,
-            borderJoinStyle: 'miter',
-            pointBorderColor: 'rgba(75,192,192,1)',
-            pointBackgroundColor: '#fff',
-            pointBorderWidth: 1,
-            pointHoverRadius: 5,
-            pointHoverBackgroundColor: 'rgba(75,192,192,1)',
-            pointHoverBorderColor: 'rgba(220,220,220,1)',
-            pointHoverBorderWidth: 2,
-            pointRadius: 1,
-            pointHitRadius: 10,
-            data: data.reverse(),
-          },
-        ],
+        datasets: [populateChart('AAPL', data)],
       };
-      this.setState({ chartData });
+      setChartData(chart);
     });
-  }
-
-  render() {
-    if (!this.state.chartData) {
-      return null;
-    } else {
-      return (
-        <div>
-          <div
-            style={{
-              width: '90vw',
-              height: '60vh',
-              margin: '0 auto',
-            }}>
-            <Line
-              data={this.state.chartData}
-              options={{
-                scales: {
-                  xAxes: [
-                    {
-                      ticks: {
-                        autoSkip: true,
-                        maxTicksLimit: 20,
-                      },
+  }, []);
+  if (!chartData) {
+    return null;
+  } else {
+    return (
+      <div>
+        <div
+          style={{
+            width: '90vw',
+            height: '60vh',
+            margin: '0 auto',
+          }}>
+          <Line
+            data={chartData}
+            options={{
+              scales: {
+                xAxes: [
+                  {
+                    ticks: {
+                      autoSkip: true,
+                      maxTicksLimit: 20,
                     },
-                  ],
-                },
-              }}
-            />
-          </div>
+                  },
+                ],
+              },
+            }}
+          />
         </div>
-      );
-    }
+      </div>
+    );
   }
-}
+};
+
+const populateChart = (ticker, data) => {
+  return {
+    label: ticker,
```

Comment:
i prefer a chartoptions object with the right names declared on top of file or separate file then you can just return the object